src/core/src/lib/models/fieldconfig.ts

```diff
@@ -34,7 +34,10 @@ export interface FormlyFieldConfig {
   /**
    * This is reserved for the templates. Any template-specific options go in here. Look at your specific template implementation to know the options required for this.
    */
-  templateOptions?: FormlyTemplateOptions;
+  props?: FormlyFieldProps;
```

Comment:
```ts 
export interface FormlyFieldConfig<Props extends FormlyFieldProps> {
  …
  props?: Props;
```

This could be a generic. Then you could make union types of the fields and get better type autocompletion.

E.g:
```ts
export interface TextareaFieldConfig extends FormFieldConfig<TextareaProps> {
  type: "textarea";
}

export interface TextareaProps extends FormlyFieldProps {
  row: number;
  cols: number;
}

export type FieldConfigTypes = TextareaFieldConfig | TelephoneFieldConfig | …;
```

And then I would urgently remove additionalProps as this makes accurate typing much more difficult.

For the transition I would allow these additionalProps only at `templateOptions` and not at `props`.